cassandra/connection.py

```diff
@@ -309,16 +309,17 @@ def __repr__(self):
 
 class SniEndPointFactory(EndPointFactory):
 
-    def __init__(self, proxy_address, port):
+    def __init__(self, proxy_address, port, node_domain=None):
         self._proxy_address = proxy_address
         self._port = port
+        self._node_domain = node_domain
```

Comment:
node_domain cannot be empty, it should result in exception

Reply:
Well I was trying to be backward compatible here,  assuming we try pushing some of those changes it upstream.
 
Anyhow we would fail on it missing from the config bundle, so it's leave only case someone would initiate it directly (which we only do in tests)